src/adjustproxyconnection.cpp

```diff
@@ -0,0 +1,196 @@
+
+    QByteArray header = line.trimmed();
+    if (header.isEmpty()) {
+      break;
+    }
+
+    pos = header.indexOf(":");
+    if (pos == -1) {
+      continue;
+    }
+
+    QByteArray headerName = header.left(pos);
+    header.remove(0, pos + 1);
+    QByteArray headerValue = header.trimmed();
+
+    QPair<QString, QString> headerPair;
+    headerPair.first = QString(headerName);
+    headerPair.second = QString(headerValue);
+
+    m_headers.append(headerPair);
+  }
+
+  m_state = ProcessingState::HeadersDone;
+}
+
+void AdjustProxyConnection::processParameters() {
+  if (m_method == "GET") {
+    m_parameters = QUrlQuery(m_route);
+  } else {
+    if (m_buffer.isEmpty()) {
+      return;
+    }
+    m_parameters = QUrlQuery(m_buffer.trimmed());
+  }
+
+  m_state = ProcessingState::ParametersDone;
+}
+
+void AdjustProxyConnection::filterParametersAndForwardRequest() {
+  QList<QPair<QString, QString>> newParameters;
+
+  for (QPair<QString, QString> parameter : m_parameters.queryItems()) {
+    if (parameter.first == "adid" || parameter.first == "app_token" ||
+        parameter.first == "attribution_deeplink" ||
+        parameter.first == "bundle_id" || parameter.first == "device_type" ||
+        parameter.first == "environment" || parameter.first == "event_token" ||
+        parameter.first == "idfv" ||
+        parameter.first == "needs_response_details" ||
+        parameter.first == "os_name" || parameter.first == "os_version" ||
+        parameter.first == "package name" ||
+        parameter.first == "reference_tag" ||
+        parameter.first == "tracking_enabled" ||
+        parameter.first == "zone_offset" || parameter.first == "att_status") {
+      newParameters.append(
+          QPair<QString, QString>(parameter.first, parameter.second));
+    } else {
+      newParameters.append(QPair<QString, QString>(
+          parameter.first, QByteArray(parameter.second.size(), '0')));
+    }
+  }
+
+  m_parameters.setQueryItems(newParameters);
+
+  forwardRequest();
+}
+
+void AdjustProxyConnection::forwardRequest() {
+  NetworkRequest* request;
+
+  request = NetworkRequest::createForAdjustProxy(
+      this, m_method, m_route.toString(), m_headers, m_parameters.toString());
+
+  connect(request, &NetworkRequest::requestFailed,
+          [this, request](QNetworkReply::NetworkError, const QByteArray& data) {
+            QByteArray response(HTTP_RESPONSE);
+            response.replace("<status>",
+                             QByteArray::number(request->statusCode()));
+            response.replace("<message>", data);
+            m_connection->write(response);
+            m_connection->close();
+          });
+
+  connect(request, &NetworkRequest::requestCompleted,
+          [this](const QByteArray& data) {
+            logger.debug() << "Adjust request succeeded";
+            logger.debug() << data;
+            QByteArray response(HTTP_RESPONSE);
+            response.replace("<status>", "200");
```

Comment:
`QByteArray::number(request->statusCode())`. There are many successful status code. Anything in the range 200 <> 299 is considered OK.